hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingReadHandler.java

```diff
@@ -145,14 +145,17 @@ public void requestMigration(NonBlockingIOThread newOwner) {
 
     @Override
     public void handle() {
-        eventCount.inc();
-        lastReadTime = Clock.currentTimeMillis();
-        if (!connection.isAlive()) {
-            String message = "We are being asked to read, but connection is not live so we won't";
-            logger.finest(message);
-            return;
-        }
         try {
+            eventCount.inc();
+            // we are going to set the timestamp even if the socketChannel is going to fail reading. In that case
+            // the connection is going to be closed anyway.
+            lastReadTime = Clock.currentTimeMillis();
```

Comment:
To make it simple. There was tricky try/catch stuff done with multiple blocks and makes it hard to understand. So now the try/catch is around the whole method.

It is in preparation of another pr... getting rid of the try/catch block by placing it from the outside.